components/dash-core-components/src/fragments/Graph.react.js

```diff
@@ -155,7 +154,19 @@ class PlotlyGraph extends Component {
             this._hasPlotted &&
             figure.data.length === gd.data.length
         ) {
-            return Plotly.animate(gd, figure, animation_options);
+            // in case we've have figure frames,
+            // we need to recreate frames before animation
+            let result;
+            if (figure.frames) {
+                result = Plotly.deleteFrames(gd).then(() => {
+                    return Plotly.addFrames(gd, figure.frames).then(() => {
+                        return Plotly.animate(gd, figure, animation_options);
+                    });
+                });
+            } else {
+                result = Plotly.animate(gd, figure, animation_options);
+            }
+            return result;
```

Comment:
```suggestion
            if (figure.frames) {
                return Plotly.deleteFrames(gd)
                    .then(() => Plotly.addFrames(gd, figure.frames))
                    .then(() => Plotly.animate(gd, figure, animation_options));
            }
            return Plotly.animate(gd, figure, animation_options);
```
Nice find! Just simplifying the code a little...

I wonder though, it looks inconsistent that `layoutClone` happens after this. To cover the responsive case wouldn't we want the `Plotly.animate` calls to take `{data: figure.data, layout: layoutClone}` instead of `figure`?

Reply:
Сalling the react and animate functions does not happen at the same time, but I think it makes sense to pass the same data.

Reply:
OK - I'm pretty sure only `data` and `layout` get used by `Plotly.animate` but at least `frames` was passed in before and hasn't seemed to cause any problems, `config` should be the same.